nihms-native-assembler/src/test/java/org/dataconservancy/nihms/assembler/nihmsnative/NihmsMetadataSerializerTest.java

```diff
@@ -144,4 +151,69 @@ public void testSerializedMetadataValidity() throws Exception {
 
     }
 
+    @Test
+    public void testUnmarshalMarshalIsIdentity() throws Exception {
+        XStream xstream = new XStream(new DomDriver("UTF-8", new XmlFriendlyNameCoder("_-", "_")));
+        xstream.registerConverter(new NihmsMetadataConverter());
+        XStream.setupDefaultSecurity(xstream);
+        xstream.alias("nihms-submit",NihmsMetadata.class);
+        xstream.allowTypesByWildcard(new String[] {
+            "org.dataconservancy.nihms.model.**"
+        });
+
+        InputStream is = underTest.serialize();
+        byte[] buffer = new byte[is.available()];
+        is.read(buffer);
+        is.close();
+
+        File targetFile = new File("MetadataRoundtripTest.xml");
+
+        OutputStream os = new FileOutputStream(targetFile);
+        os.write(buffer);
+        os.close();
+
+        NihmsMetadata roundtrippedMetadata = (NihmsMetadata) xstream.fromXML(new FileInputStream(targetFile));
```

Comment:
The assertions that follow in the code are great!

I'm also wondering if `NihmsMetadata` implements `Object.equals(Object)`?  Can we test that `NihmsMetadata.equals(Object)` works for checking if two instances are equal as well?  That may mean reviewing the implementation of `NihmsMetadata.equals(Object)` and adjusting as needed.

Reply:
i think some adjusting would be needed, as we haven't over-ridden the equals() method in NihmsMetadata, or in any of its inner classes. as for existing equality checks, 

assertEquals(metadata,roundtrippedMetadata);  and
assertTrue(metadata.equals(roundtrippedMetadata));

both fail in the current case. because the instances are different.

while a shorter equality check is great when things pass, having the assertions on the individual fields may be more useful when troubleshooting changes to the converter class, as these converters can get somewhat opaque.